server/src-lib/Hasura/Server/Init.hs

```diff
@@ -33,6 +33,7 @@ type RawAuthHook = AuthHookG (Maybe T.Text) (Maybe AuthHookType)
 data RawServeOptions
   = RawServeOptions
   { rsoPort          :: !(Maybe Int)
+  , rsoHost          :: !(Maybe String)
```

Comment:
Instead of `String` we can directly use `HostPreference` (http://hackage.haskell.org/package/warp-3.2.25/docs/Network-Wai-Handler-Warp.html#t:HostPreference)